pkg/apis/resource/validation/validation.go

```diff
@@ -681,3 +710,47 @@ func validateMap[K ~string, T any](m map[K]T, maxSize int, validateKey func(K, *
 	}
 	return allErrs
 }
+
+func validateDeviceStatus(device resource.AllocatedDeviceStatus, fldPath *field.Path, allocatedDevices sets.Set[structured.DeviceID]) field.ErrorList {
+	var allErrs field.ErrorList
+	deviceID := structured.DeviceID{Driver: device.Driver, Pool: device.Pool, Device: device.Device}
+	if !allocatedDevices.Has(deviceID) {
+		allErrs = append(allErrs, field.Invalid(fldPath, deviceID, "must be an allocated device in the claim"))
+	}
+	allErrs = append(allErrs, validateSlice(device.Conditions, -1,
+		func(condition metav1.Condition, fldPath *field.Path) field.ErrorList {
+			var allErrs field.ErrorList
+			return allErrs
+		}, fldPath.Child("conditions"))...)
+	allErrs = append(allErrs, validateRawExtension(device.Data, fldPath.Child("data"))...)
+	allErrs = append(allErrs, validateNetworkDeviceData(device.NetworkData, fldPath.Child("networkData"))...)
+	return allErrs
+}
+
+func validateRawExtension(rawExtension *runtime.RawExtension, fldPath *field.Path) field.ErrorList {
```

Comment:
the existing code in https://github.com/kubernetes/kubernetes/blob/91d7882e867da25ae8014f679db32b20e35e89b4/pkg/apis/resource/validation/validation.go#L238-L243

has a comment `	// Validation of RawExtension as in https://github.com/kubernetes/kubernetes/pull/125549/`, we should add the comment if this is copied from there

Reply:
Right, I added the comment, thank you.